### PR TITLE
add support for magento 2.3, change the behaviour of Transport class …

### PR DESCRIPTION
**Pull request by giraclion**
…to use new Zend\Mail library

### CODE REVIEW

File: Helper/Settings.php [hunk 1]
```diff
@@ -314,4 +314,15 @@ public function getHelperByTemplateId($templateId)
...
     }
+
+    /**
+     * To get magento version
+     *
+     * @return string
+     */
+    public function getMagentoVersion()
+    {
+        $productMetadata = $this->objectManager->get('Magento\Framework\App\ProductMetadataInterface');
```
Comment (giraclion) [inline Helper/Settings.php]: @ladonenko do not use OM here, inject ProductMetadataInterface via construct and use it here

File: composer.json [hunk 2]
```diff
@@ -3,7 +3,6 @@
   "description": "Sailthru Module for Magento",
   "type":"magento2-module",
   "require": {
-    "php": "~5.6.0|~7.0.0",
```
Comment (giraclion) [inline composer.json]: @ladonenko do not remove PHP version restrictions, instead extend the list of those versions:
```
"php": "~5.6.0|~7.0.0|~7.1.3|~7.2.0",
```

File: Helper/Settings.php (hunk 1 above)
Comment (ladonenko) [reply Helper/Settings.php]: added ProductMetadataInterface to DI

File: composer.json (hunk 2 above)
Comment (ladonenko) [reply composer.json]: done

**Pull request merged**